### PR TITLE
HPCC-31248 k8s multi-thor queue target

### PR DESCRIPTION
**Pull request by jakesmith**
Add ability for k8s Thor definitions to specify additional queues, such that multiple disparate Thor's can [also] listen to the same named queue(s).
This allows, for example, separate Thor's that are bound to different availability zone node pools, to be configured to listen to a single queue.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -1119,16 +1136,46 @@ Generate instance queue names
...
+  {{- end -}}
+  {{- $_ := set $stdThorQueues .name $queueItem -}}
+ {{- end -}}
+{{- end -}}
+{{- $auxThorQueues := dict -}}
+{{- range $thorNameWithAuxQueues, $auxQueues := $stdThorQueuesWithAux -}}
+ {{ $queueItem := get $stdThorQueues $thorNameWithAuxQueues -}}
+ {{- range $auxQueueName := $auxQueues -}}
+  {{- if (hasKey $stdThorQueues $auxQueueName) -}}
+   {{- $_ := fail (printf "Thor '%s' defines aux queue '%s' that clashes with existing Thor name" $queueItem.name $auxQueueName) -}}
```
Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:1159]: Please can you check in test cases in `testing\helm\errtests` and a good example in `testing\helm\tests`

**Review by ghalliday — COMMENTED** [review]
@jakesmith The change looks good.  Please can you also check in some regression tests.

**Comment by jakesmith** [comment]
@ghalliday - please see added testing/helm test cases.

**Review by ghalliday — APPROVED** [review]
@jakesmith looks good, please squash

**Comment by jakesmith** [comment]
@ghalliday - squashed.

**Pull request merged**